File: src/go-backend/internal/models/season.go
package models

import (
	"time"
)

// URL params
type SeasonRequest struct {
	ServiceTag   string
	ShowID       string
	SeasonNumber int
}

// Season Response
type Season struct {
	BackdropURL        string    `json:"backdropURL"`
	Number             int       `json:"number"`
	EpisodeCount       int       `json:"episodeCount"`
	Name               string    `json:"name"`
	OriginalName       string    `json:"originalName"`
	Overview           string    `json:"overview"`
	PosterURL          string    `json:"posterURL"`
	FirstAirDate       time.Time `json:"firstAirDate"`
	AvailabilityStatus string    `json:"availabilityStatus"`
	Episodes           []Episode `json:"episodes"`
}

File: src/frontend/src/lib/types/season.ts
import type { Episode } from "./episode";

export type SeasonRequest = {
    ServiceTag: string;
    ShowID: string;
    SeasonNumber: number;
};

export type Season = {
    backdropURL: string;
    number: number;
    episodeCount: number;
    name: string;
    originalName: string;
    overview: string;
    posterURL: string;
    firstAirDate: string;
    availabilityStatus: string;
    episodes: Episode[];
};
